rust/lance/src/index/vector.rs

```diff
@@ -479,6 +485,21 @@ pub(crate) async fn open_vector_index_v2(
             )?)
         }
 
+        index_type
+            if dataset
+                .session
+                .vector_index_extensions
+                .contains_key(index_type) =>
+        {
+            let ext = dataset
+                .session
+                .vector_index_extensions
+                .get(index_type)
+                .expect("the extension should exist");
+            ext.load_index(dataset.clone(), column, uuid, reader)
+                .await?
+        }
+
         _ => {
             return Err(Error::Index {
                 message: format!("Unsupported index type: {}", index_metadata.index_type),
```

Comment:
Since the alternative here is already `Err` maybe it would be more readable to just do...

```
index_type => {
  let ext = ...unwrap_or_else(|| Err(...))
}
```